tests/src/python/test_qgsfield.py

```diff
@@ -0,0 +1,69 @@
+"""QGIS Unit tests for QgsField.
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+__author__ = 'Nyall Dawson'
+__date__ = '16/08/2015'
+__copyright__ = 'Copyright 2015, The QGIS Project'
+
+import qgis  # NOQA
+from qgis.PyQt.QtCore import QVariant
+from qgis.core import (
+    Qgis,
+    QgsField,
+    QgsCoordinateReferenceSystem
+)
+from qgis.testing import start_app, unittest
+
+start_app()
+
+
+class TestQgsField(unittest.TestCase):
+
+    def test_metadata(self):
+        field = QgsField()
+        self.assertFalse(field.metadata())
+
+        # set custom metadata
+        field.setMetadata(Qgis.FieldMetadataProperty.CustomProperty + 2, 'test')
+        self.assertEqual(field.metadata(Qgis.FieldMetadataProperty.CustomProperty + 2),
+                         'test')
```

Comment:
Should we have a registry of custom properties ?
That way we can ensure not two different modules risk to use the same numbers (those naked integers make me uncomfortable)

Reply:
> Should we have a registry of custom properties

Maybe, but it's overkill for now. The enum is designed to handle common properties.